File: ProyectoMysql/server/EntityLayer/RecepcionEntity.py
from datetime import datetime
from pydantic import BaseModel
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum

class RecepcionDetalleSaveModel(BaseModel):
    RecepcionDetalleId: int = 0
    RecepcionId: int = 0
    MercaderiaId: int = 0
    Cantidad: float = 0
    Lote: str = ''
    FechaIngreso: datetime = datetime.now()
    FechaRegistro: datetime = datetime.now()
    FechaVencimiento: datetime = datetime.now()
    Observacion: str = ''
    Action: ProcessActionEnum = ProcessActionEnum.Loaded
    NomProducto: str = ''
    CodigoUM: str = ''
    
    @classmethod
    def Cargar(cls, _DB):
        return cls.parse_obj(_DB)

class RecepcionSaveModel(BaseModel):
    RecepcionId: int = 0
    ProcesoId: int = 0
    EstadoProcesoId: int = 0
    Codigo: str = ''
    EntidadId: int = 0
    ObjetoId: int = 0
    TipoComprobanteId: int = 0
    SerieComprobante: str = ''
    CorrelativoComprobante: str = ''
    FechaRecepcion: datetime = datetime.now()
    FechaRegistro: datetime = datetime.now()
    CodUsuario: str = ''
    Observacion : str = ''
    Action: ProcessActionEnum = ProcessActionEnum.Loaded
    ValorEstadoProceso : int = ''
    NomEstadoProceso : str = ''
    DetalleItems :list[RecepcionDetalleSaveModel] = []
    
    @classmethod
    def Cargar(cls, _DB):
        return cls.parse_obj(_DB)

class RecepcionMainModel(BaseModel):
    RecepcionId: int  = 0
    Codigo: str = ''
    TipoRecepcion: str = ''
    TipoComprobante: str = ''
    SerieComprobante: str = ''
    CorrelativoComprobante: str = ''
    FechaRecepcion: datetime = datetime.now()
    FechaRegistro: datetime = datetime.now()
    CodUsuario: str = ''
    ValorEstadoProceso: int = 0 
    NomEstadoProceso: str = ''
    
    @classmethod
    def Cargar(cls, _DB):
        return cls.parse_obj(_DB)

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.EntityLayer/RecepcionEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.EntityLayer
{
    public partial class RecepcionEntity : BaseEntityObject
    {
        public RecepcionEntity()
        {
            this.RecepcionId = 0;
            this.ProcesoId = 0;
            this.EstadoProcesoId = 0;
            this.Codigo = String.Empty;
            this.EntidadId = 0;
            this.ObjetoId = 0;
            this.TipoComprobanteId = 0;
            this.SerieComprobante = String.Empty;
            this.CorrelativoComprobante = String.Empty;
            this.FechaRecepcion = DateTime.Now;
            this.FechaRegistro = DateTime.Now;
            this.CodUsuario = String.Empty;
            this.Observacion = String.Empty;
            this.ValorEstadoProceso = 0;
            this.NomEstadoProceso = String.Empty;
            this.DetalleItem = new List<RecepcionDetalleEntity>();
        }

        public Int32 RecepcionId { get; set; }
        public Int32 ProcesoId { get; set; }
        public Int32 EstadoProcesoId { get; set; }
        public String Codigo { get; set; }
        public Int32 EntidadId { get; set; }
        public Int32 ObjetoId { get; set; }
        public Int32 TipoComprobanteId { get; set; }
        public String SerieComprobante { get; set; }
        public String CorrelativoComprobante { get; set; }
        public DateTime FechaRecepcion { get; set; }
        public DateTime FechaRegistro { get; set; }
        public String CodUsuario { get; set; }
        public String Observacion { get; set; }
        public List<RecepcionDetalleEntity> DetalleItem { get; set; }




    }
}
